Resources/public/js/views/services/ez-navigationhubviewservice.js

```diff
@@ -53,6 +71,95 @@ YUI.add('ez-navigationhubviewservice', function (Y) {
             });
         },
 
+        _load: function (next) {
+            var api = this.get('capi'),
+                loadError = false,
+                service = this;
+
+            api.getContentService().loadRoot(function (error, response) {
+                var rootLocationId,
+                    rootMediaFolderId,
+                    tasks = new Y.Parallel();
+
+                if ( error ) {
+                    service._error("Failed to contact the REST API");
+                    return;
+                }
+
+                rootLocationId = response.document.Root.rootLocation._href;
+                service._loadLocationAndContent(rootLocationId, tasks.add(function (error, result){
+                    if ( error ) {
+                        loadError = true;
+                        return;
+                    }
+                    service._rootStruct = result;
+                }));
+
+                rootMediaFolderId = response.document.Root.rootMediaFolder._href;
+                service._loadLocationAndContent(rootMediaFolderId, tasks.add(function (error, result){
+                    if ( error ) {
+                        loadError = true;
+                        return;
+                    }
+                    service._rootMediaStruct = result;
+                }));
+
+                tasks.done(function () {
+                    if ( loadError ) {
+                        service._error("Failed to load content/location");
```

Comment:
the error message will be showed to the end user so we have a provide a better one. *Failed to the load root contents and locations* is a bit better